driver-core/src/main/com/mongodb/internal/async/AsyncBatchCursor.java

```diff
@@ -28,9 +30,9 @@
  */
 public interface AsyncBatchCursor<T> extends Closeable {
     /**
-     * Returns the next batch of results.  A tailable cursor will block until another batch exists.  After the last batch, the next call
-     * to this method will execute the callback with a null result to indicate that there are no more batches available and the cursor
-     * has been closed.
+     * Returns the next batch of results. A tailable cursor will block until another batch exists.
+     * Unlike the {@link BatchCursor} this method will automatically mark the cursor as closed when there are no more expected results.
```

Comment:
The cursor no longer calls `callback.onResult(null, null)`. As this class is internal and consumed internally there is no public API breakage or behavour change.